libs/platform/unit/platform_unit_test.cpp

```diff
@@ -472,6 +536,43 @@ TEST_CASE("extension_test", "[platform]")
     GUID provider_module_id = {};
     REQUIRE(ebpf_guid_create(&client_module_id) == EBPF_SUCCESS);
     REQUIRE(ebpf_guid_create(&provider_module_id) == EBPF_SUCCESS);
+
+    REQUIRE(
+        ebpf_provider_load(
+            &provider_context,
+            &interface_id,
+            &provider_module_id,
+            nullptr,
+            &provider_data,
+            &test_provider_dispatch_table,
+            &callback_context,
+            (NPI_PROVIDER_ATTACH_CLIENT_FN*)test_denied_client_attach_provider,
+            (NPI_PROVIDER_DETACH_CLIENT_FN*)test_provider_detach_client,
+            nullptr) == EBPF_SUCCESS);
+    REQUIRE(
+        ebpf_provider_load(
+            &provider_context,
+            &interface_id,
+            &provider_module_id,
+            nullptr,
+            &provider_data,
+            &test_provider_dispatch_table,
+            &callback_context,
+            (NPI_PROVIDER_ATTACH_CLIENT_FN*)test_failed_provider_attach_client,
+            (NPI_PROVIDER_DETACH_CLIENT_FN*)test_provider_detach_client,
+            nullptr) == EBPF_SUCCESS);
+    REQUIRE(
```

Comment:
How are you validating that you get the correct behavior from NMR here?

Reply:
I think Alan means, that for each of the provider_load (NMR provder) you will also have to call extension_load (NMR client) - and check that it is returning the same error code that the test provider is returning.